enclave-runner/src/usercalls/mod.rs

```diff
@@ -346,37 +385,228 @@ impl EnclaveState {
             fds: Mutex::new(fds),
             last_fd,
             exiting: AtomicBool::new(false),
-            usercall_ext : usercall_ext,
+            usercall_ext,
+            work_sender: Mutex::new(work_sender)
         })
     }
 
+    fn create_worker_threads(num_of_worker_threads: usize,
+                             work_receiver: crossbeam::crossbeam_channel::Receiver<Work>,
+                             io_queue_send: mpsc::Sender<UsercallSendData>) {
+        for _ in 0..num_of_worker_threads {
+            let work_receiver = work_receiver.clone();
+            let io_queue_send = io_queue_send.clone();
+
+            thread::spawn(move || {
+                loop {
+                    let work = work_receiver.recv();
+                    let work = match work {
+                        Err(crossbeam::channel::RecvError) => {
+                            // the main thread has exited, so break
+                            break;
+                        },
+                        Ok(work) => {
+                            work
+                        },
+                    };
+                    work.do_work(&io_queue_send);
+                }
+            });
+        }
+    }
+
+    fn syscall_loop(io_queue_receive: mpsc::Receiver<UsercallSendData>, enclave: Arc<EnclaveState>) -> StdResult<(u64, u64), EnclaveAbort<EnclavePanic>> {
+        let main_result;
+        'outer: loop {
+            let maybe_block_recv = |block| if block {
+                io_queue_receive.recv().ok()
+            } else {
+                io_queue_receive.try_recv().ok()
+            };
+            'inner: while let Some((coresult, mut state, buf)) = maybe_block_recv(true) {
+                let mut erased_tcs = None;
+                let enclave = enclave.clone();
+                let my_result;
+                match coresult {
```

Comment:
Write `let my_result = match coresult { ... };`, this avoids having to write `my_result = ` everywhere.